src/gui/callstatechecker.cpp

```diff
@@ -0,0 +1,181 @@
+    stopChecking();
+    Q_EMIT notificationTimerElapsed();
+}
+
+void CallStateChecker::slotStatusCheckTimerElapsed()
+{
+    // Don't run check if another check is still ongoing
+    if (!_stateCheckJob.isNull()) {
+        return;
+    }
+
+    startCallStateCheck();
+}
+
+bool CallStateChecker::isAccountServerVersion22OrLater() const
+{
+    if(!_accountState || !_accountState->account()) {
+        return false;
+    }
+
+    const auto accountNcVersion = _accountState->account()->serverVersionInt();
+    const auto ncVersion22 = OCC::Account::makeServerVersion(22, 0, 0);
+
+    return accountNcVersion >= ncVersion22;
+}
+
+void CallStateChecker::startCallStateCheck()
+{
+    // check connectivity and credentials
+    if (!(_accountState && _accountState->isConnected() &&
+          _accountState->account() && _accountState->account()->credentials() &&
+          _accountState->account()->credentials()->ready())) {
+        qCInfo(lcCallStateChecker, "Could not connect, can't check call state.");
+        return;
+    }
+
+    // Check for token
+    if(_token.isEmpty()) {
+        qCInfo(lcCallStateChecker, "No call token set, can't check without it.");
+    }
+
+    qCInfo(lcCallStateChecker) << "Checking state of call with token: " << _token;
+
+    const auto spreedPath = QStringLiteral("ocs/v2.php/apps/spreed/");
+    const auto callApiPath = isAccountServerVersion22OrLater() ? QStringLiteral("api/v4/call/") : QStringLiteral("api/v1/call/");
+    const QString callPath = spreedPath + callApiPath + _token; // Make sure it's a QString and not a QStringBuilder
+
+    _stateCheckJob = new JsonApiJob(_accountState->account(), callPath, this);
+    connect(_stateCheckJob.data(), &JsonApiJob::jsonReceived, this, &CallStateChecker::slotCallStateReceived);
+
+    _stateCheckJob->setVerb(JsonApiJob::Verb::Get);
+    _stateCheckJob->start();
+}
+
+void CallStateChecker::slotCallStateReceived(const QJsonDocument &json, int statusCode)
+{
+    if (statusCode != successStatusCode) {
+        qCInfo(lcCallStateChecker) << "Failed to retrieve call state data. Server returned status code: " << statusCode;
+        return;
+    }
+
+    qCInfo(lcCallStateChecker, "Successfully retrieved call state data.");
+
+    const auto participantsJsonArray = json.object().value("ocs").toObject().value("data").toArray();
+    qCDebug(lcCallStateChecker) << participantsJsonArray;
+
+    if (participantsJsonArray.empty()) {
+        qCInfo(lcCallStateChecker, "Call has no participants and has therefore been abandoned.");
+        Q_EMIT callAbandoned();
+        stopChecking();
+        return;
+    }
+
+    for (const auto &participant : participantsJsonArray) {
+        const auto participantDataObject = participant.toObject();
+        const auto participantId = isAccountServerVersion22OrLater() ? participantDataObject.value("actorId").toString() : participantDataObject.value("userId").toString();
+
+        qDebug() << _accountState->account()->davUser();
```

Comment:
left from debug or would you like to keep it ?

Reply:
Nope, removed it